includes/class-wc-stripe-helper.php

```diff
@@ -315,6 +315,41 @@ public static function get_minimum_amount() {
 		return $minimum_amount;
 	}
 
+	/**
+	 * Gets the supported card brands, taking the store's base country and currency into account.
+	 * For more information, please see: https://stripe.com/docs/payments/cards/supported-card-brands.
+	 *
+	 * @since 4.9.0
+	 * @version 4.9.0
+	 * @return array
+	 */
+	public static function get_supported_card_brands() {
+		$base_country = wc_get_base_location()['country'];
+		$base_currency = get_woocommerce_currency();
+
+		$supported_card_brands = array( 'visa', 'mastercard' );
+
+		// American Express is not supported in Brazil and Malaysia.
+		if ( ! in_array( $base_country, array( 'BR', 'MY' ) ) ) {
+			array_push( $supported_card_brands, 'amex' );
+		}
+
+		// Discover and Diners Club are only supported in the US and Canada. 
+		if ( in_array( $base_country, array( 'US', 'CA' ) ) && $base_currency === 'USD' ) {
```

Comment:
Discover and Diners Club are only supported in the US and Canada in USD? I couldn't find that information on Stripe docs.

Reply:
Thanks, that's a good question. When I was looking into this, I found some third party documentation stating that Discover and Diners Club were only valid in US and Canada, however, I think that this information is now out of date. On our Stripe dashboard, it says the following (this is the only mention of it I can find anywhere on Stripe's documentation):

<img width="1194" alt="Screenshot 2021-02-24 at 09 46 48" src="https://user-images.githubusercontent.com/11770181/108982272-84763780-7685-11eb-8784-f4a2354134fe.png">

I think it would be safe to remove this check, although I wonder whether we should continue to restrict Diners Club and Discover to cases where the currency is USD as we did in the old logic? 

I think also that American Express not being supported in Brazil is no longer the case so I can also remove that check. (Although it is still unsupported in Malaysia: https://support.stripe.com/questions/declined-charges-on-malaysia-based-stripe-accounts

Reply:
Ignore this comment, as discussed over Slack, the note about BR and MY is here: https://stripe.com/docs/payments/cards/supported-card-brands and we found that with Diner's and Discover, it is limited to only US and CA stores, but only in US is it required to use USD, so the code will be amended to reflect that.